MCO-356: daemon: Do a secondary in-place update if rpm-ostree is too old

cgwalters opened the pull request:
xref https://github.com/coreos/rpm-ostree/commit/89f58028f0bea5b6fa59bdb3506078e09957ec00

On firstboot, the rpm-ostree (and skopeo) running in the system
may be too old to do the native flow.  This hacky bit runs
the new OS image *as a container*, and reboots another time before
doing anything else.

yuqi-zhang commented on pkg/daemon/daemon.go:
```diff
@@ -855,6 +856,20 @@ func (dn *Daemon) RunFirstbootCompleteMachineconfig() error {
+	// If the host isn't new enough to understand the new container model natively, run as a privileged container and reboot.
+	// See https://github.com/coreos/rpm-ostree/pull/3961 and https://issues.redhat.com/browse/MCO-356
+	if !newEnough {
+		err := runCmdSync("podman", "run", "--privileged", "--pid=host", "--net=host", "--rm", "-v", "/:/run/host", mc.Spec.OSImageURL, "rpm-ostree", "ex", "deploy-from-self", "/run/host", "--reboot")
```
Just curious, this is basically updating the on-host rpm-ostree from the updated rpm-ostree in the updated (4.12) oscontainer. Why is the reboot needed vs, say, restarting the rpm-ostree daemon?

cgwalters replied:
The apply-live path [was only stabilized in 2022.13](https://github.com/coreos/rpm-ostree/pull/3907/commits/30bf326147774b2ebbaa5abf4f9e0d2cfdee2432)...earlier versions do have `ex apply-live`, but really old versions don't.

We could indeed optimize this in the future to skip a reboot for sufficiently new versions; but I think ultimately what we want to do is to enhance this code to do all the setup from the container too, to avoid the *next* update doing a reboot.

IOW our firstboot path then always runs rpm-ostree from a container, and not the host.

On the other hand, we have some code for updating bootimages, and once that lands this code is going to be hit much less often.

cgwalters commented:
OK I've 

- Rebased this on https://github.com/openshift/machine-config-operator/pull/3317
- Dropped the dependency on https://github.com/openshift/machine-config-operator/pull/3302

/test e2e-agnostic-upgrade

yuqi-zhang commented:
I believe leaving the WIP is fine, the "draft" github status is what is preventing it

sinnykumari commented:
also commit message needs to be updated for https://github.com/openshift/machine-config-operator/pull/3310/commits/9a7cea465471a67bc04b54a17426c1d4d1f3e532

sinnykumari commented:
/test e2e-agnostic-upgrade

cgwalters commented:
> /test e2e-agnostic-upgrade

Oops sorry I hadn't noticed you had a test run going here, would have been nice to see the results!  But in any case we really need to land a PR as an atomic unit that has everything, so might as well do it in the other PR.

cgwalters commented:
This was merged in https://github.com/openshift/machine-config-operator/pull/3317
